mesonbuild/compilers/mixins/clike.py

```diff
@@ -476,6 +476,17 @@ def cross_compute_int(self, expression: str, low: T.Optional[int], high: T.Optio
             if self._compile_int(f'{expression} == {guess}', prefix, env, extra_args, dependencies):
                 return guess
 
+        # Try to expand the expresseion and evaluate it on the build machines compiler
+        expanded, _ = self.get_define(expression, prefix, env, extra_args, dependencies, False)
+        evaluate_expanded = f'''
+        #include <stdio.h>
+        #include <stdint.h>
+        int main(void) {{ int expression = {expanded}; printf("%d", expression); return 0; }}'''
+        run = env.coredata.compilers[mesonlib.MachineChoice.BUILD]['c'].run(evaluate_expanded, env)
```

Comment:
Error cross-compiling the `common: 38 alignment` test to armhf:

```
  File "/__w/meson/meson/mesonbuild/compilers/mixins/clike.py", line 485, in cross_compute_int
    run = env.coredata.compilers[mesonlib.MachineChoice.BUILD]['c'].run(evaluate_expanded, env)
KeyError: 'c'
```

This also points out another issue: you have to use the build compiler for the same language as the current compiler. Not 'c' unconditionally.

Reply:
not only that but I'll also have to check that self.language for the build machine is actually found.

Theoretically I should probably use `compilers.detect_compiler_for(env, self.language, MachineChoice.BUILD, False, <subproject>)` but I do not have subproject available at this point